### ISSUE TITLE
A question about scaling

### ISSUE THREAD

**Issue by sslavova8** [TPM034A/Q2_2023#11]
In the solutions of Lab 2, the data is scaled before performing the train-test split. When working on ML problems in practice, are we not supposed to first split the data, fit the scaler to the training data only, and then apply the same scale to the test set? So that we ensure there is no data leakage and the test set can be used to evaluate the performance of the developed model. :)

**Comment by Svenja4781** [comment]
I also have a question regarding the scaling of the features:
Is the scaling only applied to the originally numerical features or also to the categorical ones (after they have been one-hot encoded?)

**Comment by sslavova8** [comment]
I have an issue regarding my first question. 
I tried to first scale and encode the data and then split it into training and test sets. 
However, I ran into an issue that one of the categorical columns (purpose) contains two possible classes that are only present once in the dataset (K-12 School, Vocational education). So it was not possible to split the data first and then encode it because both datasets end up having different dimensions. 

What should I do in this case? Is it a problem to perform the train/test split at the end? Or should I drop the 2 records? Or an alternative workaround?

**Comment by spierenbu** [comment]
You should include only the following two categorical features: purpose and destination regions. (see question 3.1)

**Comment by trrbls** [comment]
> What should I do in this case? Is it a problem to perform the train/test split at the end? Or should I drop the 2 records? Or an alternative workaround?

I think you should first dummy-encode the categorical features and the split the data

**Comment by sslavova8** [comment]
> > What should I do in this case? Is it a problem to perform the train/test split at the end? Or should I drop the 2 records? Or an alternative workaround?
> 
> I think you should first dummy-encode the categorical features and then split the data

I managed to first split the data and then encode (and scale) it by using the LabelEncoder instead of the dummy-encode.
